test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/DictionaryModelBinderTest.cs

```diff
@@ -388,9 +388,8 @@ public class DictionaryModelBinderTest
                     { typeof(IDictionary<int, int>), true },
                     { typeof(Dictionary<int, int>), true },
                     { typeof(SortedDictionary<int, int>), true },
-                    { typeof(IList<KeyValuePair<int, int>>), false },
-                    { typeof(DictionaryWithInternalConstructor<int, int>), false },
-                    { typeof(DictionaryWithThrowingConstructor<int, int>), false },
+                    { typeof(IList<KeyValuePair<int, int>>), true },
```

Comment:
Like I said, now supports `IList<KeyValuePair<int, int>>` though that wasn't a goal (or a downside).